internal/pkg/keystores/validate.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2022 Nethermind
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package keystores
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+//go:embed keystore_schema_eip2335.json
+var keystoreJSONSchema []byte
+
+var (
+	ErrDepositDataNotFound              = errors.New("deposit_data.json not found")
+	ErrKeystorePasswordNotFound         = errors.New("keystore_password.txt not found")
+	ErrValidatorKeysDirNotFound         = errors.New("validator_keys directory not found")
+	ErrValidatorKeysDirWithoutKeystores = errors.New("validator_keys directory does not contain any keystores")
+	ErrInvalidKeystoreFile              = errors.New("invalid keystore file")
+	ErrInvalidKeystoreFileSchema        = errors.New("file does not match keystore schema (EIP-2335)")
```

Comment:
Don't commit this suggestion to avoid formating issues
```suggestion
	DepositDataNotFoundError                     = errors.New("deposit_data.json not found")
	KeystorePasswordNotFoundError           = errors.New("keystore_password.txt not found")
	ValidatorKeysDirNotFoundError              = errors.New("validator_keys directory not found")
	ValidatorKeysDirWithoutKeystoresError = errors.New("validator_keys directory does not contain any keystores")
	InvalidKeystoreFileError                            = errors.New("invalid keystore file")
	InvalidKeystoreFileSchemaError               = errors.New("file does not match keystore schema (EIP-2335)")
```